modules/distributor/receiver/shim.go

```diff
@@ -279,6 +279,7 @@ func (r *receiversShim) ConsumeTraces(ctx context.Context, td ptrace.Traces) err
 // ReportFatalError implements component.Host
 func (r *receiversShim) ReportFatalError(err error) {
 	_ = level.Error(log.Logger).Log("msg", "fatal error reported", "err", err)
+	r.StopAsync()
```

Comment:
have you tested this? i honestly don't know the services code very well. this may simply stop the receiver, but not cause the distributor to exit.

Reply:
have not yet, writing an otel receiver mock to do so

Reply:
(it doesnt work, uh one sec)

Reply:
got this working by switching the shim to be a basicservice
Will update this PR with some testing results on Monday (fed holiday tomorrow I'm off)